src/Compilers/VisualBasic/Portable/Lowering/LocalRewriter/LocalRewriter_ConditionalExpressions.vb

```diff
@@ -236,12 +231,18 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 End If
             End If
 
-            ' Optimize If(left, right) to left.GetValueOrDefault() when left is T? and right is the default value of T
             If rewrittenLeft.Type.IsNullableType() AndAlso
-               rewrittenRight.IsDefaultValue() AndAlso
                rewrittenRight.Type.IsSameTypeIgnoringAll(rewrittenLeft.Type.GetNullableUnderlyingType()) _
             Then
-                Return NullableValueOrDefault(rewrittenLeft)
+                ' Optimize If(left, right) to left.GetValueOrDefault() when left is T? and right is the default value of T
+                If rewrittenRight.IsDefaultValue() Then
+                    Return NullableValueOrDefault(rewrittenLeft)
+                End If
+
+                ' Optimize If(left, right) to left.GetValueOrDefault() when left is T? and right value is a constant of type T
+                If rewrittenRight.ConstantValueOpt IsNot Nothing OrElse rewrittenRight.Kind = BoundKind.Local OrElse rewrittenRight.Kind = BoundKind.Parameter Then
+                    Return NullableValueOrDefault(rewrittenLeft, rewrittenRight)
```

Comment:
>Return NullableValueOrDefault(rewrittenLeft, rewrittenRight) [](http://example.com/codeflow?start=20&length=60)

We shouldn't report an error when this helper is missing or otherwise bad